// @cartesi/util/dist/src/types/TestCartesiMathFactory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestCartesiMathFactory = void 0;
const contracts_1 = require("@ethersproject/contracts");
class TestCartesiMathFactory extends contracts_1.ContractFactory {
    constructor(linkLibraryAddresses, signer) {
        super(_abi, TestCartesiMathFactory.linkBytecode(linkLibraryAddresses), signer);
    }
    static linkBytecode(linkLibraryAddresses) {
        let linkedBytecode = _bytecode;
        linkedBytecode = linkedBytecode.replace(new RegExp("__\\$8b7ebe2a69e01741840a6648963dfd5409\\$__", "g"), linkLibraryAddresses["__$8b7ebe2a69e01741840a6648963dfd5409$__"]
            .replace(/^0x/, "")
            .toLowerCase());
        return linkedBytecode;
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static connect(address, signerOrProvider) {
        return new contracts_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.TestCartesiMathFactory = TestCartesiMathFactory;
const _abi = [
    {
        inputs: [
            {
                internalType: "uint256",
                name: "_num",
                type: "uint256",
            },
        ],
        name: "getLog2TableTimes1M",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
];
const _bytecode = "0x608060405234801561001057600080fd5b5061010f806100206000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c806332ef283b14602d575b600080fd5b604760048036036020811015604157600080fd5b50356059565b60408051918252519081900360200190f35b600073__$8b7ebe2a69e01741840a6648963dfd5409$__6332ef283b836040518263ffffffff1660e01b81526004018082815260200191505060206040518083038186803b15801560a957600080fd5b505af415801560bc573d6000803e3d6000fd5b505050506040513d602081101560d157600080fd5b50519291505056fea26469706673582212208cf6024f01b975b36f830b8d54bf16e5aef7b8a6f7df3086681b84a954fe7e7e64736f6c63430007040033";
