// @cartesi/descartes-sdk/dist/src/types/factories/LoggerInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoggerInterface__factory = void 0;
var ethers_1 = require("ethers");
var LoggerInterface__factory = /** @class */ (function () {
    function LoggerInterface__factory() {
    }
    LoggerInterface__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    return LoggerInterface__factory;
}());
exports.LoggerInterface__factory = LoggerInterface__factory;
var _abi = [
    {
        inputs: [
            {
                internalType: "uint64",
                name: "_log2Size",
                type: "uint64",
            },
            {
                internalType: "bytes8[]",
                name: "_data",
                type: "bytes8[]",
            },
        ],
        name: "calculateMerkleRootFromData",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "_root",
                type: "bytes32",
            },
            {
                internalType: "uint64",
                name: "_log2Size",
                type: "uint64",
            },
        ],
        name: "isLogAvailable",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
