// @cartesi/descartes-sdk/dist/src/types/factories/MachineInterface__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MachineInterface__factory = void 0;
var ethers_1 = require("ethers");
var MachineInterface__factory = /** @class */ (function () {
    function MachineInterface__factory() {
    }
    MachineInterface__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    return MachineInterface__factory;
}());
exports.MachineInterface__factory = MachineInterface__factory;
var _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "uint8",
                name: "exitCode",
                type: "uint8",
            },
        ],
        name: "StepGiven",
        type: "event",
    },
    {
        inputs: [],
        name: "getMemoryInteractor",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint64[]",
                name: "_rwPositions",
                type: "uint64[]",
            },
            {
                internalType: "bytes8[]",
                name: "_rwValues",
                type: "bytes8[]",
            },
            {
                internalType: "bool[]",
                name: "_isRead",
                type: "bool[]",
            },
        ],
        name: "step",
        outputs: [
            {
                internalType: "uint8",
                name: "",
                type: "uint8",
            },
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
